terraform/fargate-examples/sqs-dynamic-target-tracking/main.tf

```diff
@@ -0,0 +1,668 @@
+  description         = "This rule is used for update ECS Autoscaling Target BPI"
+  schedule_expression = "rate(60 minutes)"
+  
+  is_enabled = false
+
+  tags = local.tags
+}
+
+resource "aws_cloudwatch_event_target" "ecs_fargate_lambda_function" {
+  rule = aws_cloudwatch_event_rule.fargate_scaling.name
+  arn  = module.lambda_function_target_bpi_update.lambda_function_arn
+}
+
+resource "aws_cloudwatch_event_rule" "sqs_message_producer" {
+  name                = "SQSTestMessageProducer"
+  description         = "This rule is used for Send Messages to SQS Queue for testing"
+  schedule_expression = "rate(1 minute)"
+  
+  is_enabled = false
+
+  tags = local.tags
+}
+
+resource "aws_cloudwatch_event_target" "sqs_message_producer_lambda_function" {
+  rule = aws_cloudwatch_event_rule.sqs_message_producer.name
+  arn  = module.lambda_function_message_producer.lambda_function_arn
+}
+
+
+################################################################################
+# SQS Queue
+################################################################################
+
+
+module "processing_queue" {
+  source  = "terraform-aws-modules/sqs/aws"
+  version = "~> 2.0"
+
+  name = "${local.name}-processing-queue.fifo"
+  fifo_queue = true
+  content_based_deduplication = true
+  
+  tags = local.tags
+}
+
+################################################################################
+# CodePipeline
+################################################################################
+
+module "codepipeline_s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 3.0"
+
+  bucket = "codepipeline-${local.region}-${random_id.this.hex}"
+
+  # For example only - please re-evaluate for your environment
+  force_destroy = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_require_latest_tls_policy      = true
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  tags = local.tags
+}
+
+resource "aws_sns_topic" "codestar_notification" {
+  name = local.name
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid      = "WriteAccess"
+        Effect   = "Allow"
+        Action   = "sns:Publish"
+        Resource = "arn:aws:sns:${local.region}:${data.aws_caller_identity.current.account_id}:${local.name}"
+        Principal = {
+          Service = "codestar-notifications.amazonaws.com"
+        }
+      },
+    ]
+  })
+
+  tags = local.tags
+}
+
+module "codebuild_ci" {
+  source = "../../modules/codebuild"
+
+  name           = "codebuild-${local.name}"
+  service_role   = module.codebuild_ci.codebuild_role_arn
+  buildspec_path = "./application-code/container-queue-proc/templates/buildspec.yml"
```

Comment:
I think this should be ./application-code/ecsdemo-queue-proc/templates/buildspec.yml

Reply:
done